### ISSUE TITLE
CS1673 false positive

### ISSUE THREAD

**Issue by controlflow** [dotnet/roslyn#69907]
**Version Used**: 

main branch

**Steps to Reproduce**:

```
record struct RS(int param)
{
  public void InstanceMethod()
  {
    var unused03 = () => nameof(param.ToString);
  }
}
```

**Expected Behavior**:

Successful compilation, no primary ctor parameter is actually being captured into closure.

**Actual Behavior**:

False positive error:

> error CS1673: Anonymous methods, lambda expressions, query expressions, and local functions inside structs cannot access instance members of 'this'. Consider copying 'this' to a local variable outside the anonymous method, lambda expression, query expression, or local function and using the local instead.

**Comment by jjonescz** [comment]
This doesn't work in C# 11 but works in C# 12 with feature "instance member in 'nameof'". See [SharpLab](https://sharplab.io/#v2:EYLgtghglgdgPgJwKYGMD2CAmACAzgFwQFcV9sAlAZQApYyAHCBCMASgFgAoAby+2wACAZkEAWbAEkYBCDBRIAsknwALNJmodO/Xtv7YAbk2xEYRXEkwAGEQF5sm7LYB82GCyRoAZtUbMwAHQAKmiUhLAA5qwA3HzYAL5c8UA===).

**Comment by jcouv** [comment]
Closing as fixed in C# 12. Sorry, this is an instance where we were not able to provide a fine-grained LangVer diagnostic.
